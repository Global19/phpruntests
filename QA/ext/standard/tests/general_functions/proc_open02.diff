008+   bool(false)
009+   ["signaled"]=>
009-   ["signaled"]=>
010-   bool(false)
016+   int(1)
016-   int(0)
020+ bool(false)
020- bool(true)
029+   bool(false)
029-   bool(true)
035+   int(0)
035-   int(15)