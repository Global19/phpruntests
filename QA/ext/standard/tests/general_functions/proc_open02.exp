bool(true)
array(8) {
  ["command"]=>
  string(28) "/usr/bin/nohup /bin/sleep 50"
  ["pid"]=>
  int(%d)
  ["running"]=>
  bool(true)
  ["signaled"]=>
  bool(false)
  ["stopped"]=>
  bool(false)
  ["exitcode"]=>
  int(-1)
  ["termsig"]=>
  int(0)
  ["stopsig"]=>
  int(0)
}
bool(true)
array(8) {
  ["command"]=>
  string(28) "/usr/bin/nohup /bin/sleep 50"
  ["pid"]=>
  int(%d)
  ["running"]=>
  bool(false)
  ["signaled"]=>
  bool(true)
  ["stopped"]=>
  bool(false)
  ["exitcode"]=>
  int(-1)
  ["termsig"]=>
  int(15)
  ["stopsig"]=>
  int(0)
}
Done!