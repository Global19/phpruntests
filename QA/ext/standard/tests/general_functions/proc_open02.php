<?php
$ds = array(array('pipe', 'r'));

$cat = proc_open(
	'/usr/bin/nohup /bin/sleep 50',
	$ds,
	$pipes
);

sleep(1); // let the OS run the nohup process before sending the signal

var_dump(proc_terminate($cat, 1)); // send a SIGHUP
sleep(1);
var_dump(proc_get_status($cat));

var_dump(proc_terminate($cat)); // now really quit it
sleep(1);
var_dump(proc_get_status($cat));

proc_close($cat);

echo "Done!\n";

?>
