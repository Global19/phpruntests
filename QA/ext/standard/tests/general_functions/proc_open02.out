bool(true)
array(8) {
  ["command"]=>
  string(28) "/usr/bin/nohup /bin/sleep 50"
  ["pid"]=>
  int(11792)
  ["running"]=>
  bool(false)
  ["signaled"]=>
  bool(true)
  ["stopped"]=>
  bool(false)
  ["exitcode"]=>
  int(-1)
  ["termsig"]=>
  int(1)
  ["stopsig"]=>
  int(0)
}
bool(false)
array(8) {
  ["command"]=>
  string(28) "/usr/bin/nohup /bin/sleep 50"
  ["pid"]=>
  int(11792)
  ["running"]=>
  bool(false)
  ["signaled"]=>
  bool(false)
  ["stopped"]=>
  bool(false)
  ["exitcode"]=>
  int(-1)
  ["termsig"]=>
  int(0)
  ["stopsig"]=>
  int(0)
}
Done!